### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by PhilRounds** [shimat/opencvsharp#198]
I'm developing in Visual Studio Community 2015, C#

I'm trying to process about 1000 pictures sequentially, but I always crash at 240-250.  Eliminating all but the basics, I have

public Mat origImage;

OpenFileDialog ofd = new OpenFileDialog();

.. use ofd to select a list of jpegs to process...

```c#
MyMethod()
{
  for (int i = 0; i < ofd.FileNames.Count() ; i++)
            {
                try
                {
                    origImage = Cv2.ImRead(ofd.FileNames[i], ImreadModes.Color);
                }
                catch (System.TypeInitializationException ex1)
                {                    MessageBox.Show("Exception: " + ex1.Message);                }
             }
}
```

This always crashes somewhere around 240-250 pictures.  In the debugger diagnostic tools  I can watch the process memory grow until it reaches about 4G, then it crashes with an OpenCVSharp.OpenCVException "u != 0"

Any ideas / suggestions?

**Comment by Cunk** [comment]
Try calling origImage.Dispose() after you're done processing each one.

**Comment by PhilRounds** [comment]
----- Original Message -----

**Comment by Cunk** [comment]
Strange.  I performed a quick test before I posted that and without calling Dispose() I had exactly the same problem as you when looping through an image set -- same error message.  After adding Dispose() my memory stayed essentially flat I was able to loop through all 2000 images I had in my test folder.

However my test was confined to a single button click event (using a local Mat variable) and it looks like your code is a class method acting on a member variable but I wouldn't expect that to matter.   Maybe try recreating my test on your system to see if it works differently.
